packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundService.kt

```diff
@@ -0,0 +1,155 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.amplify.amplify_push_notifications
+
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.util.Log
+import androidx.core.app.JobIntentService
+import com.google.firebase.messaging.RemoteMessage
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.view.FlutterCallbackInformation
+import io.flutter.view.FlutterMain
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.collections.ArrayDeque
+
+
+class PushNotificationBackgroundService : MethodChannel.MethodCallHandler, JobIntentService() {
+
+    /**
+     * The work queue
+     */
+    private val queue = ArrayDeque<List<Any>>()
+
+    /**
+     * The Background Channel that is used to send messages to the callback dispatcher
+     */
+    private lateinit var backgroundChannel: MethodChannel
+
+    companion object {
+
+        private const val TAG = "PushBackgroundService"
+
+        @JvmStatic
+        private val JOB_ID = UUID.randomUUID().mostSignificantBits.toInt()
+
+        @JvmStatic
+        private var backgroundFlutterEngine: FlutterEngine? = null
+
+        @JvmStatic
+        private val serviceStarted = AtomicBoolean(false)
+
+        @JvmStatic
+        fun enqueueWork(context: Context, work: Intent) {
+            enqueueWork(context, PushNotificationBackgroundService::class.java, JOB_ID, work)
+        }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        startPushNotificationService(this)
+    }
+
+    private fun startPushNotificationService(context: Context) {
+        synchronized(serviceStarted) {
+            if (backgroundFlutterEngine == null) {
+
+                val callbackHandle = context.getSharedPreferences(
+                    PushNotificationConstants.SHARED_PREFERENCES_KEY, Context.MODE_PRIVATE
+                ).getLong(
+                    PushNotificationConstants.CALLBACK_DISPATCHER_HANDLE_KEY, 0
+                )
+                if (callbackHandle == 0L) {
+                    Log.w(TAG, "Warning: no callback registered")
```

Comment:
When and why would this happen?

Reply:
If the Callback Dispatcher's handle was not persisted in Shared Preference. Updated the warning message.